Framework/Algorithms/src/FilterByTime.cpp

```diff
@@ -25,81 +26,112 @@ using DataObjects::EventWorkspace_const_sptr;
 using DataObjects::EventWorkspace_sptr;
 using Types::Core::DateAndTime;
 
+namespace {
+namespace PropertyNames {
+const std::string INPUT_WKSP("InputWorkspace");
```

Comment:
"InputWorkspace" (and the other strings) won't change with successive visits to the algorithm. Also,`"InputWorkspace"` and `"OutputWorkspace"` are _de facto_ reserved keywords in the context of Mantid algorithms. There's an implicit convention in  Mantid algorithms to use the property names instead of representative local variables, so I strongly suggest sticking to the convention of the code base. There's also the added benefit that the property names, such as `AbsoluteStartTime`, are more informative than their variable names counterpart, such as `ABS_START`.

Reply:
Correct, these are constants in an anonymous namespace. This is so you don't have to sort out the spelling of strings for property names to make sure that `init()`/`valididateInputs()`/`exec()` all use the same parameter names. Since the `const std::string` are in an anonymous namespace they are very likely optimized out by the compiler anyhow. I think `PropertyNames::INPUT_WKSP` is more readable than`"InputWorkspace"` in the same way that `M_PI` is preferred over `3.14....`. This replaces a magic value with a named concept. It also is a slowly growing convention found in
```sh
$  rg namespace.PropertyName Framework/ -t cpp -l
Framework/Crystal/src/PredictFractionalPeaks.cpp
Framework/DataHandling/src/ApplyDiffCal.cpp
Framework/DataHandling/src/LoadDiffCal.cpp
Framework/DataHandling/src/LoadNexusMonitors2.cpp
Framework/DataHandling/src/SetSample.cpp
Framework/DataHandling/src/LoadDetectorsGroupingFile.cpp
Framework/WorkflowAlgorithms/src/AlignAndFocusPowder.cpp
Framework/Algorithms/src/CorrectTOFAxis.cpp
Framework/Algorithms/src/LorentzCorrection.cpp
Framework/Algorithms/src/LineProfile.cpp
Framework/Algorithms/src/Scale.cpp
Framework/Algorithms/src/IntegrateEPP.cpp
Framework/Algorithms/src/CreateEPP.cpp
Framework/Algorithms/src/WorkflowAlgorithmRunner.cpp
Framework/Algorithms/src/ApplyTransmissionCorrection.cpp
Framework/Algorithms/src/CalculateEfficiency2.cpp
Framework/Algorithms/src/FitPeaks.cpp
```
and a common thing I do to modernize algorithms when I touch them.

You'll also notice that cross-parameter checking got moved from `exec` into, a previously non-existing, `validateInputs()`.

Reply:
You're cheating with your example: there's no finite representation of the number PI as a sequence of numbers so you _must_ use a named concept like `M_PI`.
I like more `"InputWorkspace"` because it's closer to human language than `PropertyNames::INPUT_WKSP` but I won't wage a holy war on the subject.